docs/proposals/suggestion.md

```diff
@@ -0,0 +1,408 @@
+# Suggestion CRD Design Proposal
+
+Table of Contents
+=================
+
+   * [Suggestion CRD Design Proposal](#suggestion-crd-design-proposal)
+   * [Table of Contents](#table-of-contents)
+      * [Background](#background)
+      * [Goals](#goals)
+      * [Non-Goals](#non-goals)
+      * [Design](#design)
+         * [Kubernetes API](#kubernetes-api)
+         * [GRPC API](#grpc-api)
+         * [Workflow](#workflow)
+            * [Example](#example)
+
+Created by [gh-md-toc](https://github.com/ekalinin/github-markdown-toc)
+
+## Background
+
+## Goals
+
+## Non-Goals
+
+## Design
+
+### Kubernetes API
+
+```go
+type SuggestionSpec struct {
+	//Name of the algorithm
+	AlgorithmName string `json:"algorithm_name"`
+
+	// Number of suggestions requested
+	Suggestions int `json:"suggestions"`
+
+	//Algorithm settings set by the user in the experiment config
+	AlgorithmSettings []AlgorithmSetting `json:"algorithm_settings,omitempty"`
+}
+
+type AlgorithmSetting struct {
+	Name  string `json:"name,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+type SuggestionStatus struct {
+	// Suggestion results
+	Assignments []TrialAssignment `json:"assignments,omitempty"`
+
+	Conditions []SuggestionCondition `json:"conditions,omitempty"`
+	// include all common fields
+
+}
+
+type TrialAssignment struct {
+	// Suggestion results
+	Assignments []ParameterAssignment `json:"assignments,omitempty"`
+	Name        string                `json:"name,omitempty"`
+}
+
+type ParameterAssignment struct {
+	Name  string `json:"name,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+type Suggestion struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   SuggestionSpec   `json:"spec,omitempty"`
+	Status SuggestionStatus `json:"status,omitempty"`
+}
+```
+
+### GRPC API
+
+```protobuf
+syntax = "proto3";
+
+package api.v1.alpha3;
+
+import "google/api/annotations.proto";
+
+service AdvisorSuggestion {
+    rpc GetSuggestions(GetAdvisorSuggestionsRequest) returns (GetAdvisorSuggestionsReply);
+}
+
```

Comment:
do we need any other changes in GRPC API?

Reply:
I think we need. Thus I will have a detailed design for the API this week.